### PR TITLE
Fix for S3 watcher errors.

### PR DESCRIPTION
**Pull request by mikegrima**
Fix for #506 .  

I would like to add a test case for this before merging it in.

### CODE REVIEW

File: security_monkey/watchers/s3.py [hunk 1]
```diff
@@ -58,10 +59,19 @@ def slurp_items(**kwargs):
...
                 bucket = self.process_bucket(bucket_name, name=bucket_name, **kwargs)
-                if bucket:
+
+                # Need to check accessing the bucket was successful:
+                if bucket and not bucket.get("Error"):
                     item = S3Item.from_slurp(bucket_name, bucket, **kwargs)
                     item_list.append(item)
 
+                # If there was an error, add it to the exception map:
+                if bucket.get("Error"):
```
Comment (monkeysecurity) [inline security_monkey/watchers/s3.py]: Is it cleaner to pull the exception map and fill it directly, or to just throw the exception and let the iter_account_region decorator handle it?  Either way works.

File: security_monkey/watchers/s3.py [hunk 2]
```diff
@@ -58,6 +62,7 @@ def slurp_items(**kwargs):
 
             for bucket_name in bucket_names:
                 bucket = self.process_bucket(bucket_name, name=bucket_name, **kwargs)
+
                 if bucket:
```
Comment (cstewart87) [inline security_monkey/watchers/s3.py]: Just now running into issues here.  Appears that bucket, on L66, is always `None`.  This looks to be caused by the removal of the `return` object in the `process_bucket` def above.

File: security_monkey/watchers/s3.py [hunk 3]
```diff
@@ -41,12 +42,15 @@ def list_buckets(self, **kwargs):
...
         app.logger.debug("Slurping {index} ({name}) from {account}".format(
             index=self.i_am_singular,
-            name=bucket,
+            name=bucket_name,
             account=kwargs['account_number']))
-        return get_bucket(bucket, **kwargs)
+        bucket = get_bucket(bucket_name, **kwargs)
+
+        if bucket and bucket.get("Error"):
+            raise SecurityMonkeyException("S3 Bucket: {} fetching error: {}".format(bucket_name, bucket["Error"]))
```
Comment (cstewart87) [inline security_monkey/watchers/s3.py]: re: comments below, maybe just add this? 
```
else:
    return bucket
```

File: security_monkey/watchers/s3.py (hunk 2 above)
Comment (mikegrima) [inline security_monkey/watchers/s3.py]: Oh crap

File: security_monkey/watchers/s3.py (hunk 3 above)
Comment (mikegrima) [inline security_monkey/watchers/s3.py]: Added: #545

**Pull request merged**